### ISSUE TITLE
Error during ScheduledRefresh when partitionGranularity is enabled in preAggregation config in multi-tenancy setup  

### ISSUE THREAD

**Issue by SamShiSS** [cube-js/cube.js#638]
**Describe the bug**
see title.

The error messages look like this:
`Error querying db
Error: relation "survey_responses_main_rollup20200420_i22j0otp_odlplw4m_15893073" already exists`
`Error while querying
error: relation "survey_responses_main_rollup20200420_i22j0otp_odlplw4m_15893073" already exists`
`Refresh Scheduler Error
Error: relation "survey_responses_main_rollup20200420_i22j0otp_odlplw4m_15893073" already exists`

**To Reproduce**
1. Set up a backend using cubejs server and Postgres, with multi-tenancy setup, where each tenant's data is stored a separate Postgres Schema.
2. Create a cube with its `sql` attribute dependent on `authInfo` which varies for different tenants (since their data is stored in different Postgres Schemas)
3. Enable the `preAggregation` in the cube's options, and enable the `partitionGranularity` attribute in `preAggregation`'s options
4. Set up `cubeServer.runScheduledRefresh()` to execute at any time interval (preferably shorter like 30 seconds so that you can see the error message sooner, but not too short)
5. Set up the `cubeLogger` option for the cubejs server to log cubejs's events in the console
6. Run the backend server with Postgres
7. See error message in the console during scheduled refresh
8. If you use the cubejs client to query the server, about half the time the queries return "relation already exists" 

**Expected behavior**
No error in the logs. Querying cubejs returns actual results

**Version:**
@cubejs-backend/server: 0.19.18
@cubejs-backend/postgres-driver: 0.19.17,
postgres: 12

**Additional context**

**Comment by paveltiunov** [comment]
@SamShiSS Hey Sam! Thanks for posting this! Could you please provide your `index.js` -- where you setup multi-tenancy.

**Comment by SamShiSS** [comment]
Here it is, let me know if you need anything else
`    const server = new CubejsServer({
        devServer: process.env.NODE_ENV === 'production',
        schemaPath:
            process.env.NODE_ENV === 'production' ? 'dist/cubejs-schema' : 'src/cubejs-schema',
        basePath: '/cubejs-api',
        logger: cubeJsLogger,
        checkAuthMiddleware: (req, res, next) =>
            httpLogMiddleware(req, res, err => {
                if (err) {
                    next(err);
                    return;
                }
                checkAuthMiddleware(req, res, next);
            }),
        contextToAppId: ({ authInfo }: { authInfo: AuthInfo }) =>
            `CUBEJS_ACCOUNT_${authInfo.accountId}`,

        // Use one postgres connection pool for all tenants.
        contextToDataSourceId: () => 'common-connection-pool',

        preAggregationsSchema: ({ authInfo }: { authInfo: AuthInfo }) =>
            `acc_${authInfo.accountId.toLowerCase()}_preagg`,
        schemaVersion: ({ authInfo }: { authInfo: AuthInfo }) => {
            const version = `${authInfo.accountId}_${schemaVersions[authInfo.accountId]}`;
            return version;
        },
        queryTransformer: (query, { authInfo }) => {
            cubeApiQueryCount.inc();
            return query;
        },
    });`

basically, I intend to have one tenancy for each `accountId`

**Comment by SamShiSS** [comment]
I have found out the cause of the error. The length of the identifiers I created for the cube and its preAggregation was too long that the resulting table name exceeded Postgres's limit (63 char).
The cube's name was `SurveyResponses` and the preAggregation's name was `mainRollup`, which results in a name: survey_responses_main_rollup_20200525_o1gnpf4n_tclhjae1_1590758823820 which exceeded 63 characters and got truncated
